spdx/parser.go

```diff
@@ -40,9 +39,6 @@ type licenseID string
 
 func newLicenseID(s string) (licenseID, error) {
 	needle := s
-	if strings.HasSuffix(s, "+") {
-		needle = s[:len(s)-1]
-	}
```

Comment:
what is this change about? there's no matching test change either?

Reply:
I'll push a test update for that. With this block, a license such as `GPL-2.0++` would be considered valid.